src/client/driver/driver.js

```diff
@@ -426,6 +444,139 @@ export default class Driver extends serviceUtils.EventEmitter {
         childIframeDriverLink.sendConfirmationMessage(id);
     }
 
+    _getTargetWindowNotFoundResult () {
+        return Promise.resolve({
+            success: false,
+            errCode: TEST_RUN_ERRORS.targetWindowNotFoundError
+        });
+    }
+
+    _getChildWindowValidateResult (arr) {
+        const success = arr.find(item => item.result.success);
+
+        if (success)
+            return success.result;
+
+        let errItem = arr.find(item => item.result.errCode === TEST_RUN_ERRORS.cannotCloseWindowWithChildrenError);
+
+        if (!errItem)
+            errItem = arr.find(item => !!item.result.errCode);
+
+        return errItem ? { errCode: errItem.result.errCode } : void 0;
+    }
+
+    _handleWindowValidation (msg, wnd, windowFoundHandler, WindowValidationMessageCtor) {
+        let promiseResult = null;
```

Comment:
```js
let result = null

```

Reply:
let call not `promiseResult` or `result` but just `promise`, because real result is inside the `sendConfirmationMessage` method

Reply:
ok